tmail-backend/jmap/extensions-es-v7/src/main/java/com/linagora/tmail/james/jmap/ContactMappingFactory.java

```diff
@@ -0,0 +1,150 @@
+import static org.apache.james.backends.es.v7.IndexCreationFactory.TOKENIZER;
+import static org.apache.james.backends.es.v7.IndexCreationFactory.TYPE;
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+
+import java.io.IOException;
+
+import org.apache.james.backends.es.v7.ElasticSearchConfiguration;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+public class ContactMappingFactory {
+
+    public static final String ACCOUNT_ID = "accountId";
+    public static final String DOMAIN = "domain";
+    public static final String CONTACT_ID = "contactId";
+    public static final String EMAIL = "email";
+    public static final String FIRSTNAME = "firstname";
+    public static final String SURNAME = "surname";
+    public static final String EMAIL_AUTO_COMPLETE_ANALYZER = "email_ngram_filter_analyzer";
+    public static final String NAME_AUTO_COMPLETE_ANALYZER = "name_edge_ngram_filter_analyzer";
+    public static final String REBUILT_KEYWORD_ANALYZER = "rebuilt_keyword";
+    public static final String NGRAM_FILTER = "ngram_filter";
+    public static final String EDGE_NGRAM_FILTER = "edge_ngram_filter";
+    public static final String FILTER = "filter";
+    public static final String STANDARD = "standard";
+    public static final String TEXT = "text";
+    public static final String LOWERCASE = "lowercase";
+    public static final String MIN_GRAM = "min_gram";
+    public static final String MAX_NGRAM = "max_gram";
+
+    private final ElasticSearchConfiguration elasticSearchConfiguration;
+    private final ElasticSearchContactConfiguration contactConfiguration;
+
+    public ContactMappingFactory(ElasticSearchConfiguration configuration, ElasticSearchContactConfiguration contactConfiguration) {
+        this.elasticSearchConfiguration = configuration;
+        this.contactConfiguration = contactConfiguration;
+    }
+
+    public XContentBuilder generalContactIndicesSetting() throws IOException {
+        return jsonBuilder()
+            .startObject()
+                .startObject("settings")
+                    .field("number_of_shards", elasticSearchConfiguration.getNbShards())
+                    .field("number_of_replicas", elasticSearchConfiguration.getNbReplica())
+                    .field("index.write.wait_for_active_shards", elasticSearchConfiguration.getWaitForActiveShards())
+                    .startObject("index")
+                        .field("max_ngram_diff", contactConfiguration.getMaxNgramDiff())
+                    .endObject()
+                    .startObject("analysis")
+                        .startObject(ANALYZER)
+                            .startObject(EMAIL_AUTO_COMPLETE_ANALYZER)
+                                .field(TOKENIZER, "uax_url_email")
+                                .startArray(FILTER)
+                                    .value(NGRAM_FILTER)
+                            .endArray()
+                            .endObject()
```

Comment:
Should we add a `lowercase` filter for email? IMO not cause I see normally emails are stored in lowercase already.

Reply:
nothing prevents to have upper cases in a mail I think?

Reply:
Ok so let add a `lowercase` filter here